tests/network/test_availability.py

```diff
@@ -0,0 +1,109 @@
+import pytest_twisted as pt
+import requests
+from flask import Response
+from twisted.internet import threads
+
+from nucypher.network.sensors import AvailabilitySensor
+from nucypher.utilities.sandbox.ursula import start_pytest_ursula_services
+
+
+@pt.inlineCallbacks
+def test_availability_sensor_success(blockchain_ursulas):
+
+    # Start up self-services
+    ursula = blockchain_ursulas.pop()
+    start_pytest_ursula_services(ursula=ursula)
+
+    ursula._availability_sensor = AvailabilitySensor(ursula=ursula)
+
+    def measure():
+        ursula._availability_sensor.start()
+        assert ursula._availability_sensor.score == 10
+        ursula._availability_sensor.record(False)
+        assert ursula._availability_sensor.score == 9.0
+        for i in range(7):
+            ursula._availability_sensor.record(True)
+        assert ursula._availability_sensor.score > 9.5
+
+    def maintain():
+        sensor = ursula._availability_sensor
+        sensor.maintain()
+
+        # The node goes offline for some time...
+        for _ in range(10):
+            ursula._availability_sensor.record(False, reason={'error': 'fake failure reason'})
+
+        assert sensor.score < 4
+
```

Comment:
just for testing purposes, we could also include calls to sensor.status() next to checking of score values. eg
```suggestion
        assert not sensor.status()
        
```
in this case `4` which is less than the sensitivity threshold, hence false.

Alternatively, to be more flexible (if say the threshold changes later on) you could do the following:
```suggestion
        assert sensor.status() == (sensor.score > (sensor.SENSOR_SENSITIVITY * sensor.MAXIMUM_SCORE))
        
```

This can be repeated in other spots as well.